sandbox/visualizations_emp.R

```diff
@@ -11,176 +11,368 @@ library(leaflet.providers)
 
+#### run image-level metadata extraction for the example EXIF files ####
 
+# Get a list of the files containing the test EXIF data (one file per image dataset). These files
+# have already been created and saved into the project data folder.
+exif_files = list.files(file.path(datadir, "exif-examples"), pattern = "^exif.+\\.csv$", full.names = TRUE)
 
+# run extraction on test EXIF file
 
+exif_file1 = exif_files[1]
+exif1 = prep_exif(exif_file1)
 
+metadata1 <- extract_metadata_emp(exif_file1)
 
+# Run extraction on all EXIF files
 
+exif_file2 = exif_files[2]
+exif2 = prep_exif(exif_file2)
+metadata2 <- extract_metadata_emp(exif_file2)
 
-#### load ofo-r package ####
+exif_file3 = exif_files[3]
+exif3 = prep_exif(exif_file3)
+metadata3 <- extract_metadata_emp(exif_file3)
 
-setwd("C:\\Users\\emily\\OneDrive\\Desktop\\FOCAL\\ofo-r")
+exif_file4 = exif_files[4]
+exif4 = prep_exif(exif_file4)
+metadata4 <- extract_metadata_emp(exif_file4)
 
-# Load all the functions (and package dependencies) of this R package
-devtools::load_all()
+exif_file5 = exif_files[5]
+exif5 = prep_exif(exif_file5)
+metadata5 <- extract_metadata_emp(exif_file5)
 
-# Define the root of the local data directory
-datadir = readLines(file.path("sandbox", "data-dirs", "emp-metadata-laptop.txt"))
+exif_file6 = exif_files[6]
+exif6 = prep_exif(exif_file6)
+metadata6 <- extract_metadata_emp(exif_file6)
 
-#### run image-level metadata extraction for the example exif files ####
+#### merge metadata to original EXIF files-- later in the map creation process we need the DateTimeOriginal column instead of the new one we created in the metadata ####
 
-# Get a list of the files containing the test EXIF data (one file per image dataset). These files
-# have already been created and saved into the project data folder.
-exif_files = list.files(file.path(datadir, "exif-examples"), pattern = "^exif.+\\.csv$", full.names = TRUE)
+# first we need to create a column in the exif data.frames that can be used to match each row of data with the correct image metadata. let's choose the image path because we know each image has a unique file pathway
 
-# Define which test EXIF file to run the functions on
-#exif_file = exif_files[1]
+exif1$received_image_path <- extract_received_image_path(exif1)
 
-# Run for test EXIF file.
-#extract_metadata_emp(exif_file)
+exif2$received_image_path <- extract_received_image_path(exif2)
 
-# Run extraction on all EXIF files
-image_metadata <- purrr::map_dfr(exif_files, extract_metadata_emp)
+exif3$received_image_path <- extract_received_image_path(exif3)
 
+exif4$received_image_path <- extract_received_image_path(exif4)
 
-exif_file = exif_files[1]
+exif5$received_image_path <- extract_received_image_path(exif5)
 
-metadata1 <- ofo::extract_metadata_emp(exif_file)
+exif6$received_image_path <- extract_received_image_path(exif6)
 
+# now let's merge the exif files with the metadata files
```

Comment:
It's unclear to me why you can't just use the metadata objects. Why do they have to be joined to the exif objects? If there are additional attributes you need out of the exif objects, I would write more `extract_*` functions and add them to `extract_metadata_emp` to get them into the `metadata` objects.

Reply:
OK after reading on, I think I understand now! You need some attributes, like date/time, in a format that is specific to the leaflet mapping, and not needed as metadata otherwise. OK I think I'm back on board! One idea, (not sure if it's good) is instead of calling `extract_metadata_emp` to create a comprehensive `metadata` object, what about just creating one with the specific attribute you need, by calling the `extract_*` functions directly? If you just need a few attributes, this might be more effeicient that extracting *all* the metadata that we could possibly report. I'd advocate for switching to this because ultimately, I think our overall `extract_metadata` function will be pretty slow because of functionality not needed here (e.g., one step will be to download a DEM to see how well the drone flight tracked the terrain).

Reply:
Yes, precisely. Leaflet doesn't like the date/time format from the metadata extraction function, and I couldn't manipulate that value to get what I needed, so I resorted to working with the original exif "DateTimeOriginal" value. I don't think we need to add anything new to the extract_metadata_emp set of functions-- on the contrary I think might be confusing to have multiple date/time formats.

Reply:
I can definitely change the code so that I'm calling individual extract_* functions rather than the whole extract_metadata_emp function! I think I only need 5 or so of those extracted values. I will still have to merge them together and include the date/time calculations that rely on the original exif.